### PR TITLE
add get_or_set_hash to sdb

### LINKED ISSUE #39763: Create a way to set or get hash options on sdb
### Description of Issue/Question
Much like grains.set_or_get_hash, support using sdb.

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Allows for generating random secrets in the state files and storing them centrally, instead of having to pull the hash from grains in the mine.

### What issues does this PR fix or reference?
Resolves https://github.com/saltstack/salt/issues/39763

### Tests written?

No

### CODE REVIEW

File: salt/utils/sdb.py [hunk 1]
```diff
@@ -109,3 +114,31 @@ def sdb_delete(uri, opts, utils=None):
 
     loaded_db = salt.loader.sdb(opts, fun, utils=utils)
     return loaded_db[fun](query, profile=profile)
+
+
+def sdb_get_or_set_hash(uri,
+                        length=8,
+                        chars='abcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*(-_=+)',
+                        opts,
```
Comment (cachedout) [inline salt/utils/sdb.py]: This needs to go before the first kwarg. Please move it up one line.

File: salt/utils/sdb.py [hunk 2]
```diff
@@ -109,3 +114,31 @@ def sdb_delete(uri, opts, utils=None):
...
+                        utils=None):
+    '''
+    Check if value exists in sdb.  If it does, return, otherwise generate a
+    random string and store it.  This can be used for storing secrets in a
+    centralized place.
+    '''
+    if not isinstance(uri, string_types):
+        return False
+
+    if not uri.startswith('sdb://'):
```
Comment (cachedout) [inline salt/utils/sdb.py]: This could be combined with the previous conditional.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by gtmanfred** [comment]
Woops, my bad, should be cleaned up now.

**Comment by cachedout** [comment]
One more lint error and we should be set.

**Pull request merged**